server/api/update_user.py

```diff
@@ -5,12 +5,75 @@
 from flask import request
 from flask_restplus import Resource, fields
 from server.api import token_required
-from server.app import api
+from server.app import api, db
+from server.models.Users import Users
+
+#input form
+update_form = api.model('Update form', {
+    'user_name': fields.String,
+    'email': fields.String,
+    'pref_timezone': fields.String
+})
+
+#output form
+status = api.model('Status', {
+    'message': fields.String,
+})
 
 NS = api.namespace('update_user', description='Update a user\'s information')
 
 @NS.route('')
 class UpdateUser(Resource):
+  #What you post
+  @NS.expect(update_form)
+  #What you return
+  @NS.marshal_with(status)
+  #Checking if the user is registered
   @token_required
-  def post(self, cur_user):
-    return cur_user.to_dict()
+  def patch(self, cur_user):
+    # If cur_user if not authenticated, decorator return to the client without entering this function
+    user_request = request.get_json()  # ImmutableMultiDict
+    if user_request == None:
+      user_request = request.form
+
+    # Checking the fields
+    keys = ['user_name', 'email', 'pref_timezone']
+    for key in keys:
+      if key not in user_request:
+        return {'message': 'Request missing field: {}'.format(key), 'authenticated': False}
+
+    user_info = {
+      "id": cur_user.id,
+      "user_name": cur_user.user_name,
+      "email": cur_user.email,
+      "pref_timezone": cur_user.pref_timezone
+    }
+
+    # Extracting parameters to update
+    user_patch = {}
+    for key in user_request.keys():
+      if (user_request.get(key) == ''):
+        continue
+      if (user_info.get(key) == user_request.get(key)):
+        continue
+      user_patch[key] = user_request.get(key)
+
+    # Nothing to update
+    if (len(user_patch) == 0):
+      return {'message': 'No changes Needed'}
+
+    # Get all users
+    users = Users.query.all()
+    # Check duplicates
+    for user in users:
```

Comment:
This is going to be really slow. This would be faster: `if Users.query.filter_by(user_name=user_patch['user_name']).count() > 0:`